waspc/src/Wasp/Node/Version.hs

```diff
@@ -1,42 +1,69 @@
 module Wasp.Node.Version
-  ( getAndCheckNodeVersion,
-    getNodeVersion,
-    nodeVersionRange,
-    latestMajorNodeVersion,
-    waspNodeRequirementMessage,
-    makeNodeVersionMismatchMessage,
+  ( getAndCheckUserNodeVersion,
+    VersionCheckResult (..),
+    oldestWaspSupportedNodeVersion,
+    parseNodeVersionOutput,
+    isRangeInWaspSupportedRange,
   )
 where
 
+import Control.Arrow (left)
 import Data.Conduit.Process.Typed (ExitCode (..))
+import Data.Function ((&))
 import System.IO.Error (catchIOError, isDoesNotExistError)
 import qualified System.Process as P
-import Text.Read (readMaybe)
-import qualified Text.Regex.TDFA as R
+import qualified Text.Parsec as P
 import qualified Wasp.SemanticVersion as SV
+import qualified Wasp.SemanticVersion.Version as SV
+import qualified Wasp.SemanticVersion.VersionBound as SV
 import Wasp.Util (indent)
 
--- | Gets the installed node version, if any is installed, and checks that it
+-- | Wasp supports any node version equal or greater to this version.
+-- | We usually keep this one equal to the latest LTS.
+-- NOTE: If you change this value, make sure to also update it on some other places:
+--   - /.github/workflows/ci.yaml -> actions/setup-node -> node-version
+--   - /waspc/.nvmrc
+--   - /web/docs/introduction/getting-started.md -> "Requirements" section.
+oldestWaspSupportedNodeVersion :: SV.Version
+oldestWaspSupportedNodeVersion = SV.Version 18 0 0
```

Comment:
We should probably set a date for switching to Node.js 20, maybe the end of maintenance period of Node.js 18?

![image](https://github.com/wasp-lang/wasp/assets/2223680/17e39ab6-e866-4343-805f-8c83f990231e)

Reply:
Makes sense, I added it to the calendar!